ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -411,4 +419,25 @@ public void clearCookies(com.facebook.react.bridge.Callback callback) {
     return getReactApplicationContext()
         .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class);
   }
+
+  /**
+   * prefetch image to the fresco image disk cache
+   * @param url
+   */
+  @ReactMethod
+  public void prefetchImage(String url) {
+    Uri uri = null;
+    try{
+      if(url != null) {
+        uri = Uri.parse(url);
+      }
+    }catch (Exception e){
+      //ignore malformed url
+    }
+    if(uri == null){
+      return;
+    }
```

Comment:
I think you can rewrite this as:

    try {
      if (url != null && !url.isEmpty) {
        Uri.parse(url);
      }
    } catch (Exception ignored) {
      return;
    }

But Fresco probably parses the URL anyway? What happens if you pass an invalid URL to `ImageRequestBuilder` or `prefetchToDiskCache`?

If Fresco already handles this by doing nothing, you could simply do:

    if (url == null || url.isEmpty) {
      return;
    }

I'm curious - should we log a warning when people pass an empty URL to be prefetched? Seems like that would only happen by mistake and it would be a good developer experience to tell the developer about it. Look for usages of `FLog.w` throughout the codebase.